### ISSUE TITLE
Limiting instead of Allowing hosts

### ISSUE THREAD

**Issue by mottosso** [pyblish/pyblish-base#194]
### Goal

To support any host simply.
### Motivation

With [Host Registration](https://github.com/pyblish/pyblish/issues/177) on the way, a simpler and less altering alternative is to turn the tables around and use the `hosts` attribute to _limit_ when a plug-in runs, as opposed to allowing it. For example, at the moment, a plug-in may signal support for a host such as `maya`, and if Maya is running, then the plug-in runs. But, if an unknown host is run Pyblish throws a fit (an exception) and doesn't allow anything to be published.

I spoke with @larsbijl about it and he raised a good point, that it's (sometimes) better to be restrictive as opposed to forgiving as it provides better feedback in terms of things going wrong. This proposal would effectively "muddle the water" in that you now have to look for why plug-ins _are_ running, when you didn't expect them to. As opposed to look for why they _aren't_ running when you expected them to.

Given that the latter requires more upfront knowledge about Pyblish before one can get started using it - i.e. heightens the learning curve - I'd like to propose turning the `hosts` attribute into a limiting filter similar to `families`.
### Implementation

`current_host()` currently raises an exception in the event of an unknown host. This proposal would have it return a string of `unknown` instead.

A small advantageous side-effect of this is that we could produce plug-ins that run _only_ in the event of an unknown host.

``` python
import pyblish.api

class OnlyInUnknown(pyblish.api.Plugin):
  hosts = ["unknown"]
  def process(self):
    print("It's an unsupported host, but I'll go ahead and publish anyway.")
```
### Future

The change is completely backwards compatible and aligns with #177. `current_hosts()` still returns an empty list.

**Comment by tokejepsen** [comment]
+1

**Comment by mottosso** [comment]
Implemented in 1.1.1
